pkg/cache/v2/resource.go

```diff
@@ -132,3 +154,25 @@ func GetResourceReferences(resources map[string]types.Resource) map[string]bool
 	}
 	return out
 }
+
+// HashResources will take a list of resources and create a SHA256 hash sum out of the marshaled bytes
+func HashResources(resources []types.Resource) (string, error) {
+	var source []byte
```

Comment:
I think it might be cleaner and more efficient to stream the serialized bytes to the hasher as they are serialized.

```
hasher := sha256.New()

buffer := proto.NewBuffer(nil)
buffer.SetDeterministic(true)

for _, resource := range resources {

    err := buffer.Marshal(resource)
    if err != nil {
        return "",  err
    }

    hasher.Write(buffer.Bytes())

    buffer.Reset()   
}

return hex.EncodeToString(hasher.Sum(nil)), nil
```
This is analogous to reading a file a chunk at a time vs. reading a file into memory.  Hashers don't need all the bytes at once, they just need their current state which is the result of all bytes that have passed through.

Reply:
Good call, I've been debating taking this out of the resource file now because it isn't entirely needed. I originally had the idea that if users want to hash their resources and use that as their version for the new deltaStream they could use this. @kyessenov or @jessicayuen might want to weigh in on that